catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/util/CatroidTestRunner.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2017 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.util;
+
+import android.app.Application;
+import android.content.Context;
+import android.support.test.runner.AndroidJUnitRunner;
+
+public class CatroidTestRunner extends AndroidJUnitRunner {
```

Comment:
ive found some tutorials where the injection is done in a custom ActivityTestRule instead of writing a custom runner.
since we can only use one runner for the whole test setup, but could use a seperate rule for each test class, i would prefer to go with the option that provides finer granularity.
also, if something blows up in MockCatroidApplication -> MockWebRequestModule -> ... like the json beeing not there all malformed, all espresso tests would blow up, while if this is handled via the rule, only the test here that actuallly makes use of this would be affected.